frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/mention/date_transaction_handler.dart

```diff
@@ -0,0 +1,259 @@
+      getIt<ReminderBloc>().add(ReminderEvent.remove(reminderId: reminderId));
+    }
+  }
+
+  void _handleAddition(
+    BuildContext context,
+    String viewId,
+    EditorState editorState,
+    MentionBlockData data,
+    bool isPaste,
+    bool isCut,
+  ) {
+    final dateData = _MentionDateBlockData.fromData(data.$2);
+    if (dateData.dateString.isEmpty) {
+      Log.error("mention date block doesn't have a valid date string");
+      return;
+    }
+
+    if (isPaste && !isCut) {
+      _handlePasteFromCopy(
+        context,
+        viewId,
+        editorState,
+        data.$1,
+        data.$3,
+        dateData,
+      );
+    } else {
+      _handlePasteFromCut(viewId, data.$1, dateData);
+    }
+  }
+
+  void _handlePasteFromCut(
+    String viewId,
+    Node node,
+    _MentionDateBlockData data,
+  ) {
+    final dateTime = DateTime.tryParse(data.dateString);
+
+    if (data.reminderId == null || dateTime == null) {
+      return;
+    }
+
+    getIt<ReminderBloc>().add(
+      ReminderEvent.addById(
+        reminderId: data.reminderId!,
+        objectId: viewId,
+        scheduledAt: Int64(
+          data.reminderOption
+                  .getNotificationDateTime(dateTime)
+                  .millisecondsSinceEpoch ~/
+              1000,
+        ),
+        meta: {
+          ReminderMetaKeys.includeTime: data.includeTime.toString(),
+          ReminderMetaKeys.blockId: node.id,
+        },
+      ),
+    );
+  }
+
+  void _handlePasteFromCopy(
+    BuildContext context,
+    String viewId,
+    EditorState editorState,
+    Node node,
+    int index,
+    _MentionDateBlockData data,
+  ) async {
+    final dateTime = DateTime.tryParse(data.dateString);
+
+    if (data.reminderId == null || dateTime == null) {
+      return;
+    }
+
+    final reminderId = nanoid();
+    getIt<ReminderBloc>().add(
+      ReminderEvent.addById(
+        reminderId: reminderId,
+        objectId: viewId,
+        scheduledAt: Int64(
+          data.reminderOption
+                  .getNotificationDateTime(dateTime)
+                  .millisecondsSinceEpoch ~/
+              1000,
+        ),
+        meta: {
+          ReminderMetaKeys.includeTime: data.includeTime.toString(),
+          ReminderMetaKeys.blockId: node.id,
+        },
+      ),
+    );
+
+    final newMentionAttributes = {
+      MentionBlockKeys.mention: {
+        MentionBlockKeys.type: MentionType.date.name,
+        MentionBlockKeys.date: dateTime.toIso8601String(),
+        MentionBlockKeys.reminderId: reminderId,
+        MentionBlockKeys.includeTime: data.includeTime,
+        MentionBlockKeys.reminderOption: data.reminderOption.name,
+      },
+    };
+
+    // The index is the index of the delta, to get the index of the mention character
+    // in all the text, we need to calculate it based on the deltas before the current delta.
+    int mentionIndex = 0;
+    for (final (i, delta) in node.delta!.indexed) {
```

Comment:
Don't force unwrap the delta here.

Reply:
Can I early return if it's null? I thought the mention would live inside a delta

Reply:
Yeah. If the delta is null, ignore this transaction. In the common case, the mention must be in the delta. To avoid using force unwrap is to prevent crash or assertion.

Reply:
Got it, I've made the corresponding change. Thanks!